File: ex11_2.java
import java.util.Scanner;

public class ex11_2 {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        String input = sc.nextLine();
        int res = 0;
        for(int i = 0; i < input.length(); i++){
            int now = input.charAt(i) - '0';
            res = Math.max(res + now, res * now);
        }
        System.out.println(res);
    }
}

File: ex11_2.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    string str;
    cin >> str;
    int value = str[0] -'0';
    
    for(int i = 1; i < str.length(); i++){
        int now = str[i] - '0';
        value = max(value + now, value * now);
    }
    cout << value << "\n";
}